numpy/lib/function_base.py

```diff
@@ -4014,122 +4116,233 @@ def _quantile_is_valid(q):
     return True
 
 
-def _lerp(a, b, t, out=None):
-    """ Linearly interpolate from a to b by a factor of t """
-    diff_b_a = subtract(b, a)
-    # asanyarray is a stop-gap until gh-13105
-    lerp_interpolation = asanyarray(add(a, diff_b_a*t, out=out))
-    subtract(b, diff_b_a * (1 - t), out=lerp_interpolation, where=t>=0.5)
-    if lerp_interpolation.ndim == 0 and out is None:
-        lerp_interpolation = lerp_interpolation[()]  # unpack 0d arrays
+def virtual_index_formula(
+        array_size: np.array, quantiles: np.array, alpha: float, beta: float
+) -> np.array:
+    """
+   Compute the floating point indexes of an array for the linear
+   interpolation of quantiles.
+
+   Note:
+       Compared to R, -1 is added because R array start at 1 (0 for python)
+   """
+    return array_size * quantiles + (
```

Comment:
Have you looked at R source?  If so, this could be a problem since it is GPL, and so incompatible with NumPy's license.

Reply:
I've looked into several source codes including the one of R. I thought the comment would make sense because R source code for quantile was the reference (at least for me). But I can totally drop the comment it if it may cause license issues.
It doesn't add much to the comprehension anyway.

Reply:
The comment isn't the problem.  Any values lifted from source code is the problem.  In principle, anything that has been directly influenced by R source code should be removed from the PR, and a clean sheet implementation of these features would be needed to steer clear of GPL risk.  Anything the can be verified using documentation alone, or by experimenting with R software, would be OK (IANAL of course)

Ultimately I think the correct approach is really a question for @charris, @rgommers and other core maintainers about how this type of issue should be handled.

Reply:
On this specific case, the formula can be sourced with , for example, Hyndman&Fan paper "Sample Quantiles in Statistical Packages" in The American Statistician, Vol. 50, No. 4 (Nov., 1996), pp. 361-365
It is specific to quantile interpolations regardless of the language.

Reply:
I suspect this comment is here because the comment specifically mentions R, and is not actually about `virtual_index_formula`

Reply:
@bzah That is good news.  If your entire formula can be reproduced from the paper, then you can just cite the paper as the source and remove any other citations.

Reply:
If this may be 100% clean license wise, lets consider pinging the original R authors to give their OK (the R code and that paper are the authors).  I will assume the R code did not changed in relevant ways since then.